lib/pure/parsecsv.nim

```diff
@@ -72,12 +72,19 @@ proc raiseEInvalidCsv(filename: string, line, col: int,
                       msg: string) {.noreturn.} =
   var e: ref CsvError
   new(e)
+
+  # Prevent nil errors - these will not be set if file unable to be opened
+  let 
+    filename = if not filename.isNil(): filename else: ""
```

Comment:
I'm not sure I agree that the check should be there - `raiseEInvalidCsv` can be called from a number of places.  If added a check in this [open](https://github.com/singularperturbation/Nim/blob/65f689a7926bc1c6acc9771c68d232557104ff99/lib/pure/parsecsv.nim#L88), I'd have to assign `CsvParser.filename` in *this* [open](https://github.com/singularperturbation/Nim/blob/65f689a7926bc1c6acc9771c68d232557104ff99/lib/pure/parsecsv.nim#L112) as well.

I'm fine doing either (checking in both "open"s vs. checking in raiseEInvalidCsv) but I thought it would better to do nil-handling in the error or raiseEInvalidCsv procs instead.